### PR TITLE
Make all tests use Swift concurrency

### PR DESCRIPTION
**Pull request by adam-fowler**

### CODE REVIEW

File: Tests/SotoTests/Services/S3/S3Tests.swift [hunk 1]
```diff
@@ -45,336 +47,227 @@ class S3Tests: XCTestCase {
...
+                     if !TestEnvironment.isUsingLocalstack {
+                         throw error
+                     }
+                 default:
+                     throw error
+                 }
+             default:
+                 throw error
+             }
+         }*/
```
Comment (0xTim) [inline Tests/SotoTests/Services/S3/S3Tests.swift:91]: Does this need to be removed?

File: Tests/SotoTests/test.swift
Comment (0xTim) [inline Tests/SotoTests/test.swift:1]: Might be worth renaming the file to something other than just `test`

File: Tests/SotoTests/Services/S3/S3ExtensionTests.swift [hunk 2]
```diff
@@ -386,84 +237,85 @@ class S3ExtensionTests: XCTestCase {
...
-            }
-            .flatAlways { _ in
-                return S3Tests.deleteBucket(name: name, s3: s3)
-            }
-        XCTAssertNoThrow(try response.wait())
-    }
-
-    func testS3VirtualAddressing(_ urlString: String, config: AWSServiceConfig = S3ExtensionTests.s3.config) throws -> String {
+    /*
+     func testSelectObjectContent() throws {
```
Comment (0xTim) [inline Tests/SotoTests/Services/S3/S3ExtensionTests.swift:241]: Is this to be brought back in a future PR?

**Review by 0xTim — APPROVED** [review]

File: Tests/SotoTests/Services/S3/S3ExtensionTests.swift (hunk 2 above)
Comment (adam-fowler) [reply Tests/SotoTests/Services/S3/S3ExtensionTests.swift:241]: yes

File: Tests/SotoTests/Services/S3/S3Tests.swift (hunk 1 above)
Comment (adam-fowler) [reply Tests/SotoTests/Services/S3/S3Tests.swift:91]: removed

**Pull request merged**